src/vs/editor/contrib/suggest/suggestOvertypingCapturer.ts

```diff
@@ -0,0 +1,87 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { IDisposable, DisposableStore } from 'vs/base/common/lifecycle';
+import { ICodeEditor } from 'vs/editor/browser/editorBrowser';
+import { SuggestWidget } from './suggestWidget';
+
+export class OvertypingCapturer implements IDisposable {
+
+	private readonly _maxSelectionLength = 1000000;
```

Comment:
Maybe less than that `1000000`, how about `1000` to begin with

Reply:
I feel that 1000 will not be enough. Typical uses could be, for example, for wrapping a decent sized class in a namespace {}, or putting a middle chunk of code in an if statement, or wrapping to # region / # endregion. Think 50 KB will be just right?